src/Xamarin.Android.Build.Tasks/Utilities/XAJavaTypeScanner.cs

```diff
@@ -1,103 +1,91 @@
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
+using System.IO;
 
 using Java.Interop.Tools.Cecil;
+using Microsoft.Android.Build.Tasks;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
 using Mono.Cecil;
 using Xamarin.Android.Tools;
 
 namespace Xamarin.Android.Tasks;
 
-class JavaType
-{
-	public readonly TypeDefinition Type;
-	public readonly IDictionary<AndroidTargetArch, TypeDefinition>? PerAbiTypes;
-	public bool IsABiSpecific { get; }
-
-	public JavaType (TypeDefinition type, IDictionary<AndroidTargetArch, TypeDefinition>? perAbiTypes)
-	{
-		Type = type;
-		if (perAbiTypes != null) {
-			PerAbiTypes = new ReadOnlyDictionary<AndroidTargetArch, TypeDefinition> (perAbiTypes);
-			IsABiSpecific = perAbiTypes.Count > 1 || (perAbiTypes.Count == 1 && !perAbiTypes.ContainsKey (AndroidTargetArch.None));
-		}
-	}
-}
-
 class XAJavaTypeScanner
 {
-	sealed class TypeData
-	{
-		public readonly TypeDefinition FirstType;
-		public readonly Dictionary<AndroidTargetArch, TypeDefinition> PerAbi;
-
-		public bool IsAbiSpecific => !PerAbi.ContainsKey (AndroidTargetArch.None);
-
-		public TypeData (TypeDefinition firstType)
-		{
-			FirstType = firstType;
-			PerAbi = new Dictionary<AndroidTargetArch, TypeDefinition> ();
-		}
-	}
+	// Names of assemblies which don't have Mono.Android.dll references, or are framework assemblies, but which must
+	// be scanned for Java types.
+	static readonly HashSet<string> SpecialAssemblies = new HashSet<string> (StringComparer.OrdinalIgnoreCase) {
+		"Java.Interop.dll",
```

Comment:
My *guess* -- will test shortly -- is that *this* is the cause of `JavaArray<T>` being in the typemap table.

When I look at my typemap file from #8681, `Java.Interop` is *not* listed.  `TypeMapModule` from #8681 contains:

```
@.TypeMapModule.0_assembly_name = private unnamed_addr constant [23 x i8] c"Mono.Android.NET-Tests\00", align 16
@.TypeMapModule.1_assembly_name = private unnamed_addr constant [20 x i8] c"TestRunner.Core.NET\00", align 16
@.TypeMapModule.2_assembly_name = private unnamed_addr constant [19 x i8] c"Java.Interop-Tests\00", align 16
@.TypeMapModule.3_assembly_name = private unnamed_addr constant [21 x i8] c"TestRunner.NUnit.NET\00", align 16
@.TypeMapModule.4_assembly_name = private unnamed_addr constant [13 x i8] c"Mono.Android\00", align 1
@.TypeMapModule.5_assembly_name = private unnamed_addr constant [30 x i8] c"Mono.Android-Test.Library.NET\00", align 16
@.TypeMapModule.6_assembly_name = private unnamed_addr constant [30 x i8] c"Xamarin.Android.NUnitLite.NET\00", align 16
```

Reply:
Verified: if I remove this line -- the `Java.Interop.dll` entry within `SpecialAssemblies` -- then `JniTypeManagerTests.GetTypeSignature_Type()` no longer fails for me.